zcash_client_sqlite/src/testing.rs

```diff
@@ -25,19 +46,452 @@ use zcash_primitives::{
+        self.test_account
+            .as_ref()
+            .map(|(ufvk, _)| ufvk.sapling().unwrap().clone())
+    }
+
+    /// Invokes [`create_spend_to_address`] with the given arguments.
+    #[allow(deprecated)]
+    #[allow(clippy::type_complexity)]
+    pub(crate) fn create_spend_to_address(
+        &mut self,
+        usk: &UnifiedSpendingKey,
+        to: &RecipientAddress,
+        amount: Amount,
+        memo: Option<MemoBytes>,
+        ovk_policy: OvkPolicy,
+        min_confirmations: NonZeroU32,
+    ) -> Result<
+        TxId,
+        data_api::error::Error<
+            SqliteClientError,
+            commitment_tree::Error,
+            GreedyInputSelectorError<BalanceError, ReceivedNoteId>,
+            Infallible,
+            ReceivedNoteId,
+        >,
+    > {
+        create_spend_to_address(
+            &mut self.db_data,
+            &self.params,
+            test_prover(),
+            usk,
+            to,
+            amount,
+            memo,
+            ovk_policy,
+            min_confirmations,
+        )
+    }
+
+    /// Invokes [`spend`] with the given arguments.
+    #[allow(clippy::type_complexity)]
+    pub(crate) fn spend<InputsT>(
+        &mut self,
+        input_selector: &InputsT,
+        usk: &UnifiedSpendingKey,
+        request: zip321::TransactionRequest,
+        ovk_policy: OvkPolicy,
+        min_confirmations: NonZeroU32,
+    ) -> Result<
+        TxId,
+        data_api::error::Error<
+            SqliteClientError,
+            commitment_tree::Error,
+            InputsT::Error,
+            <InputsT::FeeRule as FeeRule>::Error,
+            ReceivedNoteId,
+        >,
+    >
+    where
+        InputsT: InputSelector<DataSource = WalletDb<Connection, Network>>,
+    {
+        spend(
+            &mut self.db_data,
+            &self.params,
+            test_prover(),
+            input_selector,
+            usk,
+            request,
+            ovk_policy,
+            min_confirmations,
+        )
+    }
+
+    /// Invokes [`propose_transfer`] with the given arguments.
+    #[allow(clippy::type_complexity)]
+    pub(crate) fn propose_transfer<InputsT>(
+        &mut self,
+        spend_from_account: AccountId,
+        input_selector: &InputsT,
+        request: zip321::TransactionRequest,
+        min_confirmations: NonZeroU32,
+    ) -> Result<
+        Proposal<InputsT::FeeRule, ReceivedNoteId>,
+        data_api::error::Error<
+            SqliteClientError,
+            Infallible,
+            InputsT::Error,
+            <InputsT::FeeRule as FeeRule>::Error,
+            ReceivedNoteId,
+        >,
+    >
+    where
+        InputsT: InputSelector<DataSource = WalletDb<Connection, Network>>,
+    {
+        propose_transfer::<_, _, _, Infallible>(
+            &mut self.db_data,
+            &self.params,
+            spend_from_account,
+            input_selector,
+            request,
+            min_confirmations,
+        )
+    }
+
```

Comment:
`propose_shielding` should also be supported here.

Reply:
Yeah; I only added methods that corresponded to things I was migrating out of tests, which indicates that we don't currently have a test for this.